reframe/core/logging.py

```diff
@@ -689,10 +688,22 @@ def emit(self, record):
             return
 
         try:
-            requests.post(
-                self._url, data=json_record,
-                headers=self._headers
-            )
+            while True:
+                response = requests.post(
+                    self._url, data=json_record,
+                    headers=self._headers
+                )
+                if response.status_code == 200:
+                    break
+
+                if response.status_code == 429:
+                    time.sleep(1)
```

Comment:
Since this can slow down the pipeline what about spreading the wait with a cyclic interval list, say `[.1, .2, .4, .8, 1.6, 3.2]`?